### PR TITLE
Feat- Added initial comment feature

### PR DESCRIPTION
**Pull request by mikco-hipe**
### OVERVIEW

Added initial comment feature with migration file, models, controller, form request, component and comment blade files
Modified the User and UserPostModel to include relationship
Modified post controller to insert comment section below the post
Added routes

### NOTES
Run "php artisan migrate" to update the database
### The comment feature still contains bugs.

- When a user comments, all comments will be displayed to all posts.

### SNIPPETS
image
image
image

### CODE REVIEW

File: laravel-app/app/Http/Controllers/CommentController.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use App\Models\PostComment;
+use App\Models\User;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+
+class CommentController extends Controller
+{
+    public function create(UserPost $post): View
+    {
+        return view('comments.create', compact('post'));
```
Comment (rafael-parayno) [inline laravel-app/app/Http/Controllers/CommentController.php:16]: Hmmm actually now that i noticed i dont prefer to use compact

File: laravel-app/resources/views/components/post.blade.php [hunk 2]
```diff
@@ -27,6 +27,46 @@
...
+                    $user = auth()->user();
+                @endphp
+                <div>
+                    @if (count($post->comments) > 0)
+                        <div class="comments">
+                            <h6>Comments</h6>
+
+                            @foreach ($post->comments as $comment)
+                                <div class="comment">
+                                    {{-- <p>{{ $comment->user->username}} </p> --}}
```
Comment (rafael-parayno) [inline laravel-app/resources/views/components/post.blade.php:59]: Nit remove comment

File: laravel-app/app/Http/Controllers/CommentController.php (hunk 1 above)
Comment (mikco-hipe) [reply laravel-app/app/Http/Controllers/CommentController.php:16]: oh is there a better way sir to pass data from controller to view?

File: laravel-app/resources/views/components/post.blade.php (hunk 2 above)
Comment (mikco-hipe) [reply laravel-app/resources/views/components/post.blade.php:59]: copy that sir

**Pull request closed without merging**